frontend/src/components/projects/project-form.tsx

```diff
@@ -150,216 +152,252 @@ export function ProjectForm({
-            </Button>
-          </DialogFooter>
-        </form>
+      <div className="space-y-2">
+        <Label htmlFor="setup-script">Setup Script (Optional)</Label>
+        <textarea
+          id="setup-script"
+          value={setupScript}
+          onChange={(e) => setSetupScript(e.target.value)}
+          placeholder="#!/bin/bash&#10;npm install&#10;# Add any setup commands here..."
+          rows={4}
+          className="w-full px-3 py-2 border border-input bg-background text-foreground rounded-md resize-vertical focus:outline-none focus:ring-2 focus:ring-ring"
+        />
+        <p className="text-sm text-muted-foreground">
+          This script will run after creating the worktree and before the
+          executor starts. Use it for setup tasks like installing dependencies
+          or preparing the environment.
+        </p>
+      </div>
+
+      <div className="space-y-2">
+        <Label htmlFor="dev-script">Dev Server Script (Optional)</Label>
+        <textarea
+          id="dev-script"
+          value={devScript}
+          onChange={(e) => setDevScript(e.target.value)}
+          placeholder="#!/bin/bash&#10;npm run dev&#10;# Add dev server start command here..."
+          rows={4}
+          className="w-full px-3 py-2 border border-input bg-background text-foreground rounded-md resize-vertical focus:outline-none focus:ring-2 focus:ring-ring"
+        />
+        <p className="text-sm text-muted-foreground">
+          This script can be run from task attempts to start a development
+          server. Use it to quickly start your project's dev server for testing
+          changes.
+        </p>
+      </div>
+
+      {error && (
+        <Alert variant="destructive">
+          <AlertCircle className="h-4 w-4" />
+          <AlertDescription>{error}</AlertDescription>
+        </Alert>
+      )}
+    </>
+  );
+
+  return (
+    <Dialog open={open} onOpenChange={handleClose}>
+      <DialogContent
+        className={isEditing ? 'sm:max-w-[600px]' : 'sm:max-w-[425px]'}
+      >
+        <DialogHeader>
+          <DialogTitle>
+            {isEditing ? 'Edit Project' : 'Create New Project'}
+          </DialogTitle>
+          <DialogDescription>
+            {isEditing
+              ? "Make changes to your project here. Click save when you're done."
+              : 'Choose whether to use an existing git repository or create a new one.'}
+          </DialogDescription>
+        </DialogHeader>
+
+        {isEditing ? (
+          <Tabs defaultValue="general" className="w-full -mt-2">
+            <TabsList className="grid w-full grid-cols-2 mb-4">
+              <TabsTrigger value="general">General</TabsTrigger>
+              <TabsTrigger value="templates">Task Templates</TabsTrigger>
+            </TabsList>
+            <TabsContent value="general" className="space-y-4">
+              <form onSubmit={handleSubmit} className="space-y-4">
+                {renderForm()}
```

Comment:
It is better to have this form as a component rather than a function inside its parent. Alternatively, adding useMemo() can reduce re-renders